etcdserver/etcdhttp/peer.go

```diff
@@ -30,15 +30,15 @@ const (
 )
 
 // NewPeerHandler generates an http.Handler to handle etcd peer (raft) requests.
-func NewPeerHandler(server *etcdserver.EtcdServer) http.Handler {
+func NewPeerHandler(server *etcdserver.EtcdServer, raftHandler http.Handler) http.Handler {
```

Comment:
@xiang90  its Cluster field is used inside now.
It can be changed to
`NewPeerHandler(clusterInfo etcdserver.ClusterInfo, raftHandler http.Handler)`
to get rid of *etcdserver.Server totally. Sound good?